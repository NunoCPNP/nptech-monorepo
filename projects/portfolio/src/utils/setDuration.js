export const setDuration = (minutes) => {
  const duration = new Date()

  duration.setTime(duration.getTime() + minutes * 60 * 1000)

  return duration
}
