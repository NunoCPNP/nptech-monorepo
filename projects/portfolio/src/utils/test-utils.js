import PropTypes from 'prop-types'
import { render } from '@testing-library/react'

import { ThemeProvider } from 'emotion-theming'
import { dark } from '../styles/themes'
import { ControllersProvider } from '../context/controllers'

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={dark}>
      <ControllersProvider>{children}</ControllersProvider>
    </ThemeProvider>
  )
}

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
