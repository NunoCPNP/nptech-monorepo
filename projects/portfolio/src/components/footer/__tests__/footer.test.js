import { render } from '../../../utils/test-utils'

import Footer from '../index'

describe('<Footer /> spec', () => {
  it('Should run without errors', () => {
    const { queryAllByTestId } = render(<Footer />)

    const test = queryAllByTestId(/footer/i)
    expect(test).toHaveLength(1)
  })

  it('Should match snapshot', () => {
    const { asFragment } = render(<Footer />)

    expect(asFragment(<Footer />)).toMatchSnapshot()
  })
})
