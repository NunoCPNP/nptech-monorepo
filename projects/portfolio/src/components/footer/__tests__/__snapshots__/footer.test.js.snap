// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> spec Should match snapshot 1`] = `
<DocumentFragment>
  <footer
    class="css-76r3pw"
    data-testid="footer"
  >
    <span>
      Made with love by Nuno Pereira
    </span>
  </footer>
</DocumentFragment>
`;
